apps/shell/app.js

```diff
@@ -55,9 +56,67 @@ app.use(process.env.PASSENGER_BASE_URI || '/', router);
 var server = new http.createServer(app);
 var wss = new WebSocket.Server({ noServer: true });
 
+
+
+whitelist = process.env.SSHHOST_WHITELIST;
+if (whitelist === undefined){
+  whitelist = [];
+} else {
+  whitelist = whitelist.split(':');
+}
+
+var filter = '.yml'; // add regex capability
+var files = fs.readdirSync('/etc/ood/config/clusters.d');
+files.forEach( function (file){
+  if (file.indexOf(filter) >= 0){
+    var fullPath = '/etc/ood/config/clusters.d/' + file;
+    var data = yaml.safeLoad(fs.readFileSync(fullPath)); //data is in json format
+    var errorMessage = 'Cluster config file does not follow correct v2 schema. Please see osc.github.io/ood-documentation.';
+
+    if (data.hasOwnProperty('v2')){
+      var v2 = data.v2;
+      if (v2.hasOwnProperty('metadata')){
+        var metadata = v2.metadata;
+        if (metadata.hasOwnProperty('hidden')){
+          if (!metadata.hidden){
+            if (v2.hasOwnProperty('login')){
+              var login = v2.login;
+              if(login.hasOwnProperty('host')){
+                var host = login.host;
+                var isDefault = false;
+                if(login.hasOwnProperty('default')){
+                  isDefault = login.default;
+                }
+                if (isDefault){
+                  whitelist.unshift(host);
+                } else {
+                  whitelist.push(host);
+                }
+              }
+            } else {
+              console.log(errorMessage);
+            }
+          }
+        } else {
+          console.log(errorMessage);
+        }
+      } else {
+        console.log(errorMessage);
+      }
+    } else {
+      console.log(errorMessage);
+    }
+  }
+});
+
+whitelist.push('default');
+whitelist = Array.from(new Set(whitelist)); // remove duplicates
+sshhost = whitelist[0]; //default sshhost if not set
```

Comment:
This should be named default_host or default_sshhost since it is the default ssh host.